compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/react-namespace.expect.md

```diff
@@ -29,36 +29,38 @@ import { c as _c } from "react/compiler-runtime";
 const FooContext = React.createContext({ current: null });
 
 function Component(props) {
-  const $ = _c(5);
+  const $ = _c(7);
   React.useContext(FooContext);
   const ref = React.useRef();
   const [x, setX] = React.useState(false);
   let t0;
-  if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
+  if ($[0] !== ref) {
```

Comment:
hmmm we shouldn't be taking refs as deps, this seems off?

Reply:
Should be fixed in #31521. Will rebase and update